agent.py

```
@@ -31,16 +31,157 @@
+
+        response = await client.chat.completions.create(
+            model=model,
+            messages=[{"role": "user", "content": formatted_revision_prompt}],
+            temperature=0.3, # Lower temperature for more focused, less creative revision
+            n=1,
+            stop=None,
+        )
+        if response.choices and response.choices[0].message and response.choices[0].message.content:
+            revised_task = response.choices[0].message.content.strip()
+            # Basic check to ensure LLM didn't just return an empty string or something very short
+            if len(revised_task) < 0.5 * len(prompt_text) and len(prompt_text) > 50 : # Heuristic: if significantly shorter
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)